i18n-helpers/src/bin/markdown-gettext.rs

```diff
@@ -0,0 +1,248 @@
+                if !path.is_dir() {
+                    bail!(
+                        "Specified output path exists but is not a directory: {}",
+                        path.display()
+                    );
+                }
+            } else {
+                fs::create_dir_all(&path)
+                    .with_context(|| format!("Failed to create directory: {}", path.display()))?;
+            }
+            Ok(path)
+        }
+        None => {
+            let default_path = PathBuf::from("translated_md_files");
+            fs::create_dir_all(&default_path).with_context(|| {
+                format!(
+                    "Failed to create default directory: {}",
+                    default_path.display()
+                )
+            })?;
+            Ok(default_path)
+        }
+    }
+}
+
+fn main() -> anyhow::Result<()> {
+    let args = env::args().collect::<Vec<_>>();
+
+    let mut lang_option: Option<String> = None;
+    let mut dir_option: Option<String> = None;
+    let mut file_option: Option<String> = None;
+    let mut output_path_option: Option<String> = None;
+
+    let mut iter = args.iter().peekable();
+
+    while let Some(arg) = iter.next() {
+        if arg.starts_with("--po") {
+            lang_option = split_argument(arg)
+        } else if arg.starts_with("--dir") {
+            dir_option = split_argument(arg)
+        } else if arg.starts_with("--f") {
+            file_option = split_argument(arg)
+        } else if arg.starts_with("--out") {
+            output_path_option = split_argument(arg)
+        }
+    }
+
+    let files = allocate_files(dir_option, file_option)?;
+    let catalog = build_catalog(lang_option)?;
+    let output_path = find_output_path(output_path_option)?;
+
+    translate_files(catalog, files, output_path)?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_split_argument() {
+        assert_eq!(
+            split_argument(&"--po=lang.po".to_string()),
+            Some("lang.po".to_string())
+        );
+        assert_eq!(split_argument(&"--out src/resources".to_string()), None);
+        assert_eq!(split_argument(&"--invalid".to_string()), None);
+    }
+
+    #[test]
+    fn test_validate_file() -> anyhow::Result<()> {
+        let tmp_dir = tempfile::tempdir()?;
+        let md_path = tmp_dir.path().join("test.md");
+        let html_path = tmp_dir.path().join("test.html");
+        let source_path = md_path.to_str().unwrap();
+
+        let markdown_content = "# How to Foo from a specified file path\n\
+        \n\
+        The first paragraph about Foo.\n\
+        Still the first paragraph.*baz*\n";
+
+        let html_content = "<h1>Hello Foo! </h1>";
+
+        let mut md_file = File::create(&md_path)?;
+        let mut html_file = File::create(&html_path)?;
+        write!(md_file, "{}", markdown_content)?;
+        write!(html_file, "{}", html_content)?;
+
+        assert_eq!(validate_file(&md_path), true);
```

Comment:
`cargo clippy` will most likely tell you to replace `assert_eq!(foo, true)` with `assert!(foo)` (and likewise for the `assert_eq!` below).